File: sdks/typescript/src/thinking_protocol_values.ts
/** Provider thinking wire protocols projected as local values. */
export const ThinkingProtocol = Object.freeze({
  None: "none",
  ModelManaged: "model_managed",
  OpenaiReasoningEffort: "openai_reasoning_effort",
  DeepseekReasoningEffort: "deepseek_reasoning_effort",
  AnthropicEffort: "anthropic_effort",
  AnthropicThinkingBudget: "anthropic_thinking_budget",
  AnthropicAdaptive: "anthropic_adaptive",
  EnableThinkingFlag: "enable_thinking_flag",
  ThinkingType: "thinking_type",
  GlmReasoningEffort: "glm_reasoning_effort",
  OllamaThink: "ollama_think",
} as const);

export type ThinkingProtocol = (typeof ThinkingProtocol)[keyof typeof ThinkingProtocol];

export function thinkingProtocolEmitsField(protocol: ThinkingProtocol): boolean {
  if (protocol === ThinkingProtocol.None || protocol === ThinkingProtocol.ModelManaged || protocol === ThinkingProtocol.AnthropicAdaptive) return false;
  return Object.values(ThinkingProtocol).includes(protocol);
}

File: sdks/python/src/echo_agent_sdk/thinking_protocol_values.py
from __future__ import annotations

from enum import Enum


class ThinkingProtocol(str, Enum):
    NONE = "none"
    MODEL_MANAGED = "model_managed"
    OPENAI_REASONING_EFFORT = "openai_reasoning_effort"
    DEEPSEEK_REASONING_EFFORT = "deepseek_reasoning_effort"
    ANTHROPIC_EFFORT = "anthropic_effort"
    ANTHROPIC_THINKING_BUDGET = "anthropic_thinking_budget"
    ANTHROPIC_ADAPTIVE = "anthropic_adaptive"
    ENABLE_THINKING_FLAG = "enable_thinking_flag"
    THINKING_TYPE = "thinking_type"
    GLM_REASONING_EFFORT = "glm_reasoning_effort"
    OLLAMA_THINK = "ollama_think"

    def emits_field(self) -> bool:
        return self not in {
            ThinkingProtocol.NONE,
            ThinkingProtocol.MODEL_MANAGED,
            ThinkingProtocol.ANTHROPIC_ADAPTIVE,
        }
